src/Dependencies.jl

```diff
@@ -128,6 +128,7 @@ struct Dependency <: AbstractDependency
             end
         end
         if top_level
+            @warn("Dependency(\"$(getname(pkg))\") was defined as top-level but this is deprecated, use `RuntimeDependency` instead")
```

Comment:
Strictly speaking, we could also just remove `top_level` from `Dependency` without doing a breaking release as it was never documented :innocent: